torch/nn/_functions/thnn/upsampling.py

```diff
@@ -64,13 +69,29 @@ def backward(self, grad_output):
 
 class UpsamplingBilinear2d(_UpsamplingBase):
 
+    def __init__(self, size=None, scale_factor=None):
+        super(UpsamplingBilinear2d, self).__init__(size, scale_factor)
+
+        if self.scale_factor is not None:
+            self.scale_factor = _pair(self.scale_factor)
+            # we have to be a tuple at this point
+            try:
+                assert len(self.scale_factor) == 2
+                for i in self.scale_factor:
+                    assert isinstance(i, Integral)
+                    assert i >= 1
+            except AssertionError as e:
+                raise ValueError('scale_factor must be a non-negative integer, '
+                                 'or a tuple of non-negative integers for bilinear upsamplings, but got: '
+                                 '{}'.format(self.scale_factor))
+
     def forward(self, input):
         assert input.dim() == 4
 
-        if self.scale_factor:
+        if self.scale_factor is not None:
             self.output_size = (
-                input.size(2) * self.scale_factor,
-                input.size(3) * self.scale_factor,
+                input.size(2) * self.scale_factor[0],
```

Comment:
You need to write a `__setstate__` function for modules that were serialized - `self.scale_factor` won't be a tuple for them, so you need to patch it:
```python
def __setstate__(self, state):
    self.__dict__.update(state)
    self.scale_factor = _tuple(self.scale_factor)
```